CMakeLists.txt

```diff
@@ -6,138 +6,555 @@
 #
 #===------------------------------------------------------------------------===#
 
-cmake_minimum_required(VERSION 3.9.0)
+# FIXME (?) : downgraded requirement here to match the rest of LLVM.  There 
+# were no apparent issues but perhaps future PRs will change this? 
+cmake_minimum_required(VERSION 3.4.3) 
```

Comment:
I think the version 3.9.0 reflected what the cmake file was tested with. Some old version (3.1 or 3.2 maybe) didn't work and 3.9 did. I don't think we need the FIXME comment.

Reply:
Thanks -- wasn't sure if there were other dependencies (e.g. FIR) that might have bumped this requirement up.  Will remove FIXME.

Reply:
Pat: FIR has the same dependencies as MLIR does in this regard.